src/NRedisStack/TimeSeries/DataTypes/TimeStamp.cs

```diff
@@ -23,7 +23,7 @@ public class TimeStamp
         /// Build a TimeStamp from DateTime.
         /// </summary>
         /// <param name="dateTime">DateTime value</param>
-        public TimeStamp(DateTime dateTime) => Value = dateTime.Ticks;
+        public TimeStamp(DateTime dateTime) => Value = (long) dateTime.Millisecond; // TODO: Check this
```

Comment:
Both the new and old here will produce timestamps from January 1st 1970, `Ticks` and `Millisecond` are both components of the timestamp, they do not represent the unix millisecond timestamp of `DateTime.`

You will need to use the DateTimeOffset struct to produce what is needed here:

```suggestion
        public TimeStamp(DateTime dateTime) => Value = new DateTimeOffset(dateTime).ToUnixTimeMilliseconds();
```